projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,316 @@
+
+        StringBuilder contentBuilder = new StringBuilder();
+        try {
+            BufferedReader in = new BufferedReader(new FileReader(lawsuitTemplateFilePath));
+            String str;
+            while ((str = in.readLine()) != null) {
+                contentBuilder.append(str);
+            }
+            in.close();
+        } catch (IOException e) {
+        }
+        String content = contentBuilder.toString();
+        return content;
+    }
+
+    /**
+     * Fill lawsuit template with user & spam company's details.
+     * @return Template's content as Html, with fields replaced.
+     */
+    protected String fillInTemplate() {
+
+        // Read template's content:
+        String res = templateContent;
+
+        // Fill in general form details:
+        claimCase = sentHaser ? claimCaseHaser: claimCaseSubscription;
+        fiveLawsuitsStatus = sentMoreThanFiveLawsuits ? moreThanFiveLawsuits: lessThanFiveLawsuits;
+        res = res.replace("&lt;claimCase&gt;", claimCase);
+        res = res.replace("&lt;claimAmount&gt;", claimAmount);
+        res = res.replace("&lt;spamType&gt;", spamType);
+        res = res.replace("&lt;moreThanFiveLawsuits&gt;", fiveLawsuitsStatus);
+        res = res.replace("&lt;undefined&gt;", "");
+        res = res.replace("&lt;receivedSpamDate&gt;", dateFormatter.format(receivedSpamDate));
+        res = res.replace("&lt;currentDate&gt;", dateFormatter.format(currentDate));
+
+        // User details:
+        res = res.replace("&lt;userFullName&gt;", userPrivateName + " " + userLastname);
+        res = res.replace("&lt;userId&gt;", userID);
+        res = res.replace("&lt;userAddress&gt;", userAddress);
+        res = res.replace("&lt;userPhone&gt;", userPhone);
+        res = res.replace("&lt;userFax&gt;", userFax);
+
+        // First company's details:
+        res = res.replace("&lt;companyName&gt;", companyName);
+        res = res.replace("&lt;companyId&gt;", companyId);
+        res = res.replace("&lt;companyAddress&gt;", companyAddress);
+        res = res.replace("&lt;companyPhone&gt;", companyPhone);
+        res = res.replace("&lt;companyFax&gt;", companyFax);
+
+        // Second company's details:
+        res = res.replace("&lt;company2Name&gt;", company2Name);
+        res = res.replace("&lt;company2Id&gt;", company2Id);
+        res = res.replace("&lt;company2Address&gt;", company2Address);
+        res = res.replace("&lt;company2Phone&gt;", company2Phone);
+        res = res.replace("&lt;company2Fax&gt;", company2Fax);
+
+        return res;
+    }
+
+    /**
+     * Responsible for the output lawsuit PDF file name format.
+     * @return File name with ".pdf" extension.
+     */
+    protected String createPdfLawsuitFileName(){
+
+        String fileName = dateAndTimeFormatter.format(new Date()) + ".pdf";
+        return fileName;
+    }
+
+    /**
+     * This is where configurations of folders structure of Lawsuit Activity is been done.
```

Comment:
When documenting methods - we use present simple:
https://www.grammarly.com/blog/simple-present/

Also, it's good to be short and to the point. For example, I would rewrite this as:
```
Create app folder structure.
```
Actually, now that I'm writing this, it doesn't seem to add information in addition to the method name, so we can probably just remove the comment altogether (unless I didn't notice some additional information in the comment).

Reply:
Rearranged in general other comments as well, removed some unnecessary ones.